src/Microsoft.AspNet.Mvc.Core/VirtualFilePathResult.cs

```diff
@@ -0,0 +1,134 @@
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Mvc.Core;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Internal;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// An <see cref="ActionResult"/> that when executed will write a virtual file (eg. embedded resource)
+    /// to the response using mechanisms provided by the host.
+    /// </summary>
+    public class VirtualFilePathResult : FileResult
+    {
+        private const int DefaultBufferSize = 0x1000;
+        private string _fileName;
+
+        /// <summary>
+        /// Creates a new <see cref="VirtualFilePathResult"/> instance with the provided <paramref name="fileName"/>
+        /// and the provided <paramref name="contentType"/>.
+        /// </summary>
+        /// <param name="fileName">The path to the file. The path must be relative/virtual.</param>
+        /// <param name="contentType">The Content-Type header of the response.</param>
+        public VirtualFilePathResult([NotNull] string fileName, [NotNull] string contentType)
+            : this(fileName, new MediaTypeHeaderValue(contentType))
+        {
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="VirtualFilePathResult"/> instance with
+        /// the provided <paramref name="fileName"/> and the
+        /// provided <paramref name="contentType"/>.
+        /// </summary>
+        /// <param name="fileName">The path to the file. The path must be relative/virtual.</param>
+        /// <param name="contentType">The Content-Type header of the response.</param>
+        public VirtualFilePathResult([NotNull] string fileName, [NotNull] MediaTypeHeaderValue contentType)
+            : base(contentType)
+        {
+            FileName = fileName;
+        }
+
+        /// <summary>
+        /// Gets or sets the path to the file that will be sent back as the response.
+        /// </summary>
+        public string FileName
+        {
+            get
+            {
+                return _fileName;
+            }
+
+            [param: NotNull]
+            set
+            {
+                _fileName = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="IFileProvider"/> used to resolve paths.
+        /// </summary>
+        public IFileProvider FileProvider { get; set; }
+
+        /// <inheritdoc />
+        protected override Task WriteFileAsync(HttpResponse response, CancellationToken cancellation)
+        {
+            var fileProvider = GetFileProvider(response.HttpContext.RequestServices);
+            var normalizedPath = NormalizePath(FileName);
+            var fileInfo = fileProvider.GetFileInfo(normalizedPath);
+
+            if (fileInfo.Exists)
```

Comment:
Sure, but don't the docs at the top of this class say it's for embedded resources, not physical files?
https://github.com/aspnet/StaticFiles/blob/dev/src/Microsoft.AspNet.StaticFiles/StaticFileContext.cs#L327-L328